overlay/utils.sh

```diff
@@ -114,6 +115,60 @@ dumpDepInfo() {
     done
 }
 
+install_crate2() {
+    mkdir -p "$out/lib"
+    # shellcheck disable=SC2046
+    cp $(jq -rR 'fromjson?
+        | select(.reason == "compiler-artifact")
+        | .filenames
+        | map(select(test("\\.(a|rlib|so|dylib)$")))
+        | .[]' < .cargo-build-output) "$out/lib" || :
+
+    mkdir -p "$out/bin"
+    # shellcheck disable=SC2046
+    cp $(jq -rR 'fromjson?
+        | select(.reason == "compiler-artifact")
+        | .executable
+        | select(.)' < .cargo-build-output) "$out/bin" || :
+
+    jq -rR 'fromjson?
+        | select(.reason == "build-script-executed")
+        | (.linked_libs | map("-l \(.)")) + (.linked_paths | map("-L \(.)"))
+        | .[]' < .cargo-build-output \
+        | tr '\n' ' ' >> "$out/lib/.link-flags"
+
+    local out_dir="$(jq -rR 'fromjson? | select(.reason == "build-script-executed") | .out_dir' < .cargo-build-output)"
+    touch "$out/lib/.dep-keys"
+    # `links` envs aren't included in build output so we have to manually parse them.
+    grep -P '^cargo:(?!rerun-if-changed|rerun-if-env-changed|rustc-link-lib|rustc-link-search|rustc-flags|rustc-cfg|rustc-env|rustc-cdylib-link-arg|warning)' \
+        "$out_dir/../output" | while IFS= read -r line; do
+        [[ "$line" =~ cargo:([^=]+)=(.*) ]] || continue
+        local key="${BASH_REMATCH[1]}"
+        local val="${BASH_REMATCH[2]}"
+        printf 'DEP_%s_%s=%s\n' "$(upper "$cargo_links")" "$(upper "$key")" "$val" >> "$out/lib/.dep-keys"
+    done || :
+
+    if [[ -d "$out_dir" ]] && { grep "$out_dir" "$out/lib/.dep-keys" || grep "$out_dir" "$out/lib/.link-flags"; }; then
+        local installed_out_dir="$out/build_output" # TODO: Change me
+        sed -i "s#$out_dir#$installed_out_dir#g" "$out/lib/.dep-keys"
+        sed -i "s#$out_dir#$installed_out_dir#g" "$out/lib/.link-flags"
+        cp -r "$out_dir" "$installed_out_dir"
```

Comment:
I noticed a bunch of log spam from I believe this line, where it seems to be trying to copy the `bin` folder from crates that don't have a build folder. I'll have to do a clean build to get the log output again, but see if you happen to notice that during your development as well

Reply:
Interestingly, I noticed the log spam about the failed `bin` copy when building this very PR on my machine, @edude03. Take a look at this:

```
installing
cp: missing destination file operand after '/nix/store/a2azz41rpf73mdwfzv4chiffhdh1a9jz-crate-memchr-2.2.1/bin'
Try 'cp --help' for more information.
post-installation fixup
strip is /nix/store/69in1slwg74lqz3lzz15jbrck2rik21v-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/a2azz41rpf73mdwfzv4chiffhdh1a9jz-crate-memchr-2.2.1/bin
patching script interpreter paths in /nix/store/a2azz41rpf73mdwfzv4chiffhdh1a9jz-crate-memchr-2.2.1
cp: missing destination file operand after '/nix/store/9ri7f4ygdih03npa75czwjx9qi0pcq5a-crate-opener-0.4.1/bin'
Try 'cp --help' for more information.
grep: /../output: No such file or directory
post-installation fixup
strip is /nix/store/69in1slwg74lqz3lzz15jbrck2rik21v-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/9ri7f4ygdih03npa75czwjx9qi0pcq5a-crate-opener-0.4.1/bin
patching script interpreter paths in /nix/store/9ri7f4ygdih03npa75czwjx9qi0pcq5a-crate-opener-0.4.1
cp: missing destination file operand after '/nix/store/i5p1szlajv474rclnvyyvdhsggd3mh33-crate-pathdiff-0.1.0/bin'
Try 'cp --help' for more information.
grep: /../output: No such file or directory
post-installation fixup
strip is /nix/store/69in1slwg74lqz3lzz15jbrck2rik21v-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/i5p1szlajv474rclnvyyvdhsggd3mh33-crate-pathdiff-0.1.0/bin
patching script interpreter paths in /nix/store/i5p1szlajv474rclnvyyvdhsggd3mh33-crate-pathdiff-0.1.0
cp: missing destination file operand after '/nix/store/s47ig50zjv67bynv9qhb0hprgqfy72kz-crate-percent-encoding-2.1.0/bin'
Try 'cp --help' for more information.
```

Reply:
Fixed in a095422 and 0e258d2.